### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by Yoshinonxkun**
Add latest documentation stuff, improve structure and solve some github issues

### CODE REVIEW

**Comment by Yoshinonxkun** [comment]
[Skript zum Initialimport der bestehenden Daten](https://trello.com/c/Fs8D5GS3/40-skript-zum-initialimport-der-bestehenden-daten)

**Comment by Yoshinonxkun** [comment]
[Dokumentationsaufgaben](https://trello.com/c/PngngNXn/100-dokumentationsaufgaben)

**Comment by Yoshinonxkun** [comment]
[Dokumentationsaufgaben](https://trello.com/c/PngngNXn/100-dokumentationsaufgaben)

File: docs/_sources/code/importscripts.rst.txt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Importscripts
+-------------
+
+You can use the following script/scripts to import some date in your *db.sqlite3*
+Tese script/scripts are used in the funktional tests.
```
Comment (benjamin-hempel) [inline docs/_sources/code/importscripts.rst.txt:5]: *functional

File: docs/_sources/code/importscripts.rst.txt [hunk 2]
```diff
@@ -0,0 +1,23 @@
+Importscripts
+-------------
+
+You can use the following script/scripts to import some date in your *db.sqlite3*
```
Comment (benjamin-hempel) [inline docs/_sources/code/importscripts.rst.txt:4]: *data

File: docs/_sources/code/tests.rst.txt [hunk 3]
```diff
@@ -0,0 +1,215 @@
+Tests
+-----
+
+This section contains the description of the functional UI test created with the selenium testframework. The Pagination
+and the Module historie are passively testet in the other Tests.
```
Comment (benjamin-hempel) [inline docs/_sources/code/tests.rst.txt:5]: *tested

File: docs/_sources/code/tests.rst.txt [hunk 4]
```diff
@@ -0,0 +1,215 @@
+Tests
+-----
+
+This section contains the description of the functional UI test created with the selenium testframework. The Pagination
```
Comment (benjamin-hempel) [inline docs/_sources/code/tests.rst.txt:4]: *This section contains the description of the functional UI tests created with the Selenium testing framework.

File: docs/_sources/developer/deployment.rst.txt [hunk 5]
```diff
@@ -0,0 +1,163 @@
+Deployment
+----------
+
+This section contains a step-by-step guidance how to deploy this software.
+It`s based on an deployment of Django with Apache2 and mod_wsgi on a raspberry pi 4.
```
Comment (benjamin-hempel) [inline docs/_sources/developer/deployment.rst.txt:5]: *a deployment

File: docs/_sources/developer/deployment.rst.txt [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+Install Dependencies for the deployment. (We use Python3)
+
+  ``sudo apt-get install python3-pip apache2 libapache2-mod-wsgi-py3``
+
+You also need to install virtualenv, to seperate Python from your system's python
+
+  ``sudo pip install virtualenv``
+
+Now we need to clone the Git-Repository and setup the virtualenv for Python.
+First you need to change in the directory where you want to clone this Webapplication.
```
Comment (benjamin-hempel) [inline docs/_sources/developer/deployment.rst.txt:36]: First you need to change to the directory that you want to clone this web application to.

File: docs/_sources/developer/deployment.rst.txt [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+  ``git clone https://github.com/mribrgr/StuRa-Mitgliederdatenbank.git``
+  ``cd StuRa-Mitgliederdatenbank/``
+
+Now create a virtual environment and activate it.
+
+  ``virtualenv venv``
+  ``source venv/bin/activate``
+
+If you successfully activatet your virtual ennvironement than your Promt should
```
Comment (benjamin-hempel) [inline docs/_sources/developer/deployment.rst.txt:47]: If you have successfully activated your virtual ennvironement than your prompt should

File: docs/_sources/developer/deployment.rst.txt [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+look something like this ``(venv) user@host:StuRa-Mitgliederdatenbank``. The last
+step is to install the requirements.txt in the virtual environement.
+
+  ``pip install -r requirements.txt``
+
+Adjust Django Project Settings:
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First we need to configure ``StuRa-Mitgliederdatenbank/bin/settings.py``:
+We opening the file first (based on the previous chapter)
```
Comment (benjamin-hempel) [inline docs/_sources/developer/deployment.rst.txt:57]: We open the file first (based on the previous chapter)

File: docs/_sources/developer/deployment.rst.txt [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+
+And deactivate the virtual environment
+
+  ``deactivate``
+
+Configure Apache2:
+~~~~~~~~~~~~~~~~~~
+
+To enable Apache2 as front end we need to configure WSGI pass.
+To archive this we need to edit the default virtual host file:
```
Comment (benjamin-hempel) [inline docs/_sources/developer/deployment.rst.txt:85]: *achieve

File: docs/_sources/developer/deployment.rst.txt [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+
+  ``sudo nano /etc/apache2/sites-available/000-default.conf``
+
+We can keep everything what is present in this file and add our config above
```
Comment (benjamin-hempel) [inline docs/_sources/developer/deployment.rst.txt:89]: We can keep everything that is present in this file and add our config above

File: docs/_sources/developer/deployment.rst.txt [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+    WSGIProcessGroup StuRa-Mitgliederdatenbank
+    WSGIScriptAlias / /home/pi/StuRa-Mitgliederdatenbank/bin/wsgi.py
+
+  </VirtualHost>
+
+Solve some Permission Issues:
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The first step is to change the permissions of the database, so that group owner
+can read and write. Then we need to give some files the ownership to Apache2
```
Comment (benjamin-hempel) [inline docs/_sources/developer/deployment.rst.txt:145]: Then we need to the ownership of som files to Apache2

File: docs/_sources/developer/introductionDevelopement.rst.txt [hunk 12]
```diff
@@ -0,0 +1,43 @@
+Introduction Developement
```
Comment (benjamin-hempel) [inline docs/_sources/developer/introductionDevelopement.rst.txt:1]: Development

File: docs/_sources/developer/introductionDevelopement.rst.txt [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+-------------------------
+
+This and the folowing sections deal with developement specific themes.
```
Comment (benjamin-hempel) [inline docs/_sources/developer/introductionDevelopement.rst.txt:4]: This and the following sections deal with development specific topics.

File: docs/_sources/developer/introductionDevelopement.rst.txt [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+First step is to pull the Git-Repository
+
+    ``git clone https://github.com/mribrgr/StuRa-Mitgliederdatenbank.git``
+
+Next we change the directory to the pulled Git-Repository with:
+
+    ``cd StuRa-Mitgliederdatenbank/``
+
+Now we need to install all python requirements (optional: install it in a virtualenviroment).
+You can use the requirements.txt, it contains all requirements we yoused.
```
Comment (benjamin-hempel) [inline docs/_sources/developer/introductionDevelopement.rst.txt:25]: *used

File: docs/_sources/developer/introductionDevelopement.rst.txt [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+    ``pip install -r requirements.txt``
+
+The Basics are done, now we need to create a Database and apply all migrations to it. We use the
+makemigrations and the migrate command from the manage.py.
+
+    | ``python3 ./manage.py makemigrations``
+    | ``python3 ./manage.py migrate``
+
+If the Database was successfully created there must be a db.sqlite3 in the directory.
+As last step, before we can fire up the server, we need to create with the command createsuperuser a
```
Comment (benjamin-hempel) [inline docs/_sources/developer/introductionDevelopement.rst.txt:36]: we need to create an admin account using the command superuser

File: docs/anforderung/files/csv/team.csv [hunk 16]
```diff
@@ -7,3 +7,4 @@ Schüttig, Theresa, Developer, Tester Backup
 Hempel, Benjamin, Developer, ---
 Hirsch, Lukas, Tester, Technical Writer
 Holland, Stefan, Deployment Eng., Analyst Backup
+, Marc, Architect, ---
```
Comment (benjamin-hempel) [inline docs/anforderung/files/csv/team.csv:10]: Sein voller Name ist Marc Michel

File: docs/anforderung/files/glossary.adoc [hunk 17]
```diff
@@ -40,13 +40,14 @@ Zur besseren Übersichtlichkeit sind Begriffe, Abkürzungen und Datendefinitione
...
+|LXC |Linux Containers |Virtualisierung auf Betriebssystemebene, indem mehrere Systeme isoliert gleichzeitig laufen sich aber denselben Kernel teilen.
 |===
 
 == Verzeichnis der Datenstrukturen
 [%header]
 |===
 |Bezeichnung|	Definition |	Format | Gültigkeitsregeln | Aliase
 |Anmeldedaten|Zusammensetzung von Benutzername und Passwort.|String|Emailadresse muss @-Zeichen und Punkt enthalten.|Login
-|Mitgliedsdaten |xx |xx |xx |xx 
+|Mitgliedsdaten |Zusammensetung von Name, Vorname, Legislatur(Funktion), Adresse |classe | belegte Funktion muss vorhanden sein |xx
```
Comment (benjamin-hempel) [inline docs/anforderung/files/glossary.adoc:51]: Klasse oder class

File: docs/anforderung/files/glossary.adoc [hunk 18]
```diff
@@ -40,13 +40,14 @@ Zur besseren Übersichtlichkeit sind Begriffe, Abkürzungen und Datendefinitione
...
+|Daten (funktionsspezifisch) |Referenzierung der Organisationseinheit und des Unterbereiches, der die Funktion untergeordnet ist. Zusammensetung von Bezeichnung, Workload, Maximalmitgliedern |classe |xx |xx
```
Comment (benjamin-hempel) [inline docs/anforderung/files/glossary.adoc:52]: Klasse oder class

File: docs/sphinx/source/code/importscripts.rst [hunk 19]
```diff
@@ -0,0 +1,23 @@
+Importscripts
+-------------
+
+You can use the following script/scripts to import some date in your *db.sqlite3*
```
Comment (benjamin-hempel) [inline docs/sphinx/source/code/importscripts.rst:4]: Siehe andere importscripts.rst

File: docs/sphinx/source/code/tests.rst [hunk 20]
```diff
@@ -0,0 +1,215 @@
+Tests
```
Comment (benjamin-hempel) [inline docs/sphinx/source/code/tests.rst:1]: Siehe andere tests.rst

File: docs/sphinx/source/developer/deployment.rst [hunk 21]
```diff
@@ -0,0 +1,163 @@
+Deployment
```
Comment (benjamin-hempel) [inline docs/sphinx/source/developer/deployment.rst:1]: Siehe andere deployment.rst

File: docs/sphinx/source/developer/introductionDevelopement.rst [hunk 22]
```diff
@@ -0,0 +1,43 @@
+Introduction Developement
```
Comment (benjamin-hempel) [inline docs/sphinx/source/developer/introductionDevelopement.rst:1]: Siehe andere introductionDevelopment.rst

File: importscripts/main.py [hunk 23]
```diff
@@ -8,7 +8,28 @@
...
+        * Funktion
+
+        To use this funktion you need to have a file.csv with folowing structure:
+
+        * delimiter = ','
+        * organisationseinheit,unterbereich,funktion,max_members
+        * First line is a heading and will not be imported
+
+
+        :param file: File, that contains the data which had to be importet
```
Comment (benjamin-hempel) [inline importscripts/main.py:28]: File containing the data to be imported

File: importscripts/main.py [hunk 24]
```diff
@@ -8,7 +8,28 @@
...
 def importAemter(file):
+    """
+        **!WARNING!**
+        This function clears following Tables in your Database:
+
+        * Organisationseinheit
+        * Unterbereich
+        * Funktion
+
+        To use this funktion you need to have a file.csv with folowing structure:
```
Comment (benjamin-hempel) [inline importscripts/main.py:21]: *function

File: tests/MyFuncMitglieder.py [hunk 25]
```diff
@@ -4,12 +4,12 @@
 
 def addMitglied(self):
     """
-        Hinzufüghen eines Mitglieds mit default parametern
+        Add a member with default parameters over the "mitglieder/erstellen" view.
+        You need to be on the "mitglieder/" site to call this funktion.
```
Comment (benjamin-hempel) [inline tests/MyFuncMitglieder.py:8]: *function

File: tests/MyFuncMitglieder.py [hunk 26]
```diff
@@ -102,18 +102,19 @@ def addMitgliedWithParameters(self,
                                 spitzname,
                                 ):
     """
-        Hinzufüghen eines Mitglieds mit default parametern
+        Add a member with default parameters over the "mitglieder/erstellen" view.
+        You can change firstname, lastname and username.
+        You need to be on the "mitglieder/" site to call this funktion.
```
Comment (benjamin-hempel) [inline tests/MyFuncMitglieder.py:107]: *function

File: tests/MyTestCase.py [hunk 27]
```diff
@@ -11,14 +11,27 @@
 class MyTestCase(StaticLiveServerTestCase):
     """
         Setup and Teardown funktions are specified here.
+        The following Testcases inherit from this class.
 
+        All Testcases Inherit from this class are testing the User Interface.
```
Comment (benjamin-hempel) [inline tests/MyTestCase.py:16]: All testcases inheriting from this class are testing the User Interface.

File: tests/MyTestCase.py [hunk 28]
```diff
@@ -11,14 +11,27 @@
...
     """
     # befor every test funktion
 
     def setUp(self):
         """
-            Auswahl des richtigen Webdriver anhand des Systemes
+            This Funktion is called before every testcase.
```
Comment (benjamin-hempel) [inline tests/MyTestCase.py:22]: *function

File: tests/MyTestCase.py [hunk 29]
```diff
@@ -11,14 +11,27 @@
...
+
+            It setup the webdriver and create 1 admin and 1 user.
```
Comment (benjamin-hempel) [inline tests/MyTestCase.py:24]: It sets up the webdriver and creates 1 admin and 1 user.

File: tests/MyTestCase.py [hunk 30]
```diff
@@ -64,5 +77,14 @@ def setUp(self):
 
     # after every test funktion
     def tearDown(self):
+        """
+            This Funktion is called after every testcase.
```
Comment (benjamin-hempel) [inline tests/MyTestCase.py:81]: *function

File: tests/test_001_admin.py [hunk 31]
```diff
@@ -4,14 +4,15 @@
...
+        Setup and Teardown Functions are specified in
         MyTestCase
     """
 
     # Tests
     def test_login_superuser(self):
         """
-            Hier wird getestet, ob man sich als Admin einloggen kann
+            This is a "positive" Systemtest as Blackboxtest.
+            Here we want to check if you can login as Admin and all sites are correct displayed.
```
Comment (benjamin-hempel) [inline tests/test_001_admin.py:15]: *displayed correctly

File: tests/test_002_user.py [hunk 32]
```diff
@@ -11,7 +11,9 @@ class TestUser(MyTestCase):
     # Tests
     def test_login_user(self):
         """
-            Hier wird getestet, ob man sich als User anmelden kann.
+            This is a "positive" Systemtest as Blackboxtest.
+            Here we want to check if you can login as User and all sites are correct displayed.
```
Comment (benjamin-hempel) [inline tests/test_002_user.py:15]: *displayed correctly

File: tests/test_004_mitgliedHinzufuegen.py [hunk 33]
```diff
@@ -17,17 +17,35 @@ class TestMitgliedHinzufuegen(MyTestCase):
     # Tests
     def test_1MitgliedHinzufügen_AsSuperuser(self):
         """
-            Hier wird getestet, ob man als Admin ein Mitglied Hinzufügen kann.
+            This is a "positive" Systemtest as Blackboxtest.
+            Here we want to check if you can add a new Member as Admin and if the
+            Member is correct displayed in the table.
```
Comment (benjamin-hempel) [inline tests/test_004_mitgliedHinzufuegen.py:22]: *displayed correctly

File: tests/test_004_mitgliedHinzufuegen.py [hunk 34]
```diff
@@ -17,17 +17,35 @@ class TestMitgliedHinzufuegen(MyTestCase):
...
         loginAsLukasAdmin(self)
         addMitglied(self)
         pass
 
     def test_50MitgliederHinzufügen_AsSuperuser_lookAsUser(self):
         """
-            Hier wird getestet, ob man als Admin 50 Mitglieder Hinzufügen kann und ob die Pagination Funktioniert.
+            This is a complex "positive" Systemtest as Blackboxtest.
+            Here we want to check if you can add a multiple new Members (50) as Admin and if the
+            Member is correct displayed in the table.
```
Comment (benjamin-hempel) [inline tests/test_004_mitgliedHinzufuegen.py:39]: *displayed correctly

File: tests/test_005_mitgliedEntfernen.py [hunk 35]
```diff
@@ -11,6 +11,18 @@ class TestMitgliedEntfernen(MyTestCase):
 
     # Tests
     def test_1MitgliedEntfernen_AsSuperuser(self):
+        """
+            This is a "positive" Systemtest as Blackboxtest.
+            Here we want to check if you can delete a new Member as Admin and if the
+            Member is correct deleted in the table.
```
Comment (benjamin-hempel) [inline tests/test_005_mitgliedEntfernen.py:17]: *deleted correctly

File: tests/test_006_mitgliedAendern.py [hunk 36]
```diff
@@ -11,6 +11,18 @@ class TestMitgliedAendern(MyTestCase):
 
     # Tests
     def test_1MitgliedAendern_AsSuperuser(self):
+        """
+            This is a "positive" Systemtest as Blackboxtest.
+            Here we want to check if you can change information of a Member as Admin and if the
+            Member is correct changed in the table.
```
Comment (benjamin-hempel) [inline tests/test_006_mitgliedAendern.py:17]: *changed correctly

File: tests/test_007_aemtHinzufuegen.py [hunk 37]
```diff
@@ -6,20 +6,19 @@
...
     """
-        Hier wird getestet:
-
-        * Ob ein Organisationseinheit richtig über das AdminPanel Hinzugefügt wird
-        * Ob ein Unterbereich richtig über das AdminPanel Hinzugefügt wird
-        * Ob ein Funktion richtig über das AdminPanel Hinzugefügt wird
-        * Ein Komplextest ob man mit Hinzugefügten Organisationseinheit/Unterbereich/Funktion
-
-        Ein neues Mitglied erstellen kann, und ob es in der Ämterübersicht übernommen wird
+        Setup and Teardown Funktions are specified in
```
Comment (benjamin-hempel) [inline tests/test_007_aemtHinzufuegen.py:9]: *functions

File: tests/test_007_aemtHinzufuegen.py [hunk 38]
```diff
@@ -6,20 +6,19 @@
...
+        MyTestCase
     """
 
-    def test_1ReferatHinzufuegen_AsSuperuser(self):
+    def test_1OrganisationseinheitHinzufuegen_AsSuperuser(self):
         """
-            Hier wird ein Organisationseinheit hinzugefügt und überprüft ob es dann richtig
-            übernommen wurde.
+            This is a "positive" Systemtest as Blackboxtest.
+            Here we want to check if you can add a "organisationseingeit" as Admin.
```
Comment (benjamin-hempel) [inline tests/test_007_aemtHinzufuegen.py:16]: *Organisationseinheit

File: tests/test_007_aemtHinzufuegen.py [hunk 39]
```diff
@@ -60,10 +69,19 @@ def test_1AemtHinzufuegen_AsSuperuser(self):
 
     def test_ReferatUnterbereichAmtHinzufuegen_AsSuperuser(self):
         """
-            Dies ist ein Zusammenhängender Test
-            Organisationseinheit, Unterbereich und Funktion werden neu angelegt.
-            Ein neues Mitglied wird erstellt und diesem zugeordnet.
-            Es wird Überprüft ob auch alles in die Ämterübersicht mitgenommen wird.
+            This is a complex "positive" Systemtest as Blackboxtest.
+            Here we want to check if you can add a organisationseinheit, unterbereich and funktion as Admin.
+            We also check if we can a new Member with the new data and if everything is displayed correctly
```
Comment (benjamin-hempel) [inline tests/test_007_aemtHinzufuegen.py:74]: Hier fehlt ein "add"

File: tests/test_008_aemtEntfernen.py [hunk 40]
```diff
@@ -5,14 +5,22 @@
...
-        ...
+        Setup and Teardown Funktions are specified in
+        MyTestCase
     """
 
     def test_1ReferatEntfernen_AsSuperuser(self):
         """
-            Hier wird ein Organisationseinheit hinzugefügt um es dann wieder zu entfernen.
+            This is a "positive" Systemtest as Blackboxtest.
+            Here we want to check if you can delete a "organisationseingeit" as Admin.
```
Comment (benjamin-hempel) [inline tests/test_008_aemtEntfernen.py:15]: "Organisationseinheit"

File: tests/test_008_aemtEntfernen.py [hunk 41]
```diff
@@ -5,14 +5,22 @@
 
 class TestAemtEntfernen(MyTestCase):
     """
-        Hier wird getestet:
-        ...
+        Setup and Teardown Funktions are specified in
```
Comment (benjamin-hempel) [inline tests/test_008_aemtEntfernen.py:8]: *functions

File: tests/test_008_aemtEntfernen.py (hunk 40 above)
Comment (benjamin-hempel) [inline tests/test_008_aemtEntfernen.py:15]: *Organisationseinheit

File: tests/test_009_aemtAendern.py [hunk 42]
```diff
@@ -6,17 +6,15 @@
 
 class TestAemtAendern(MyTestCase):
     """
-        Hier wird getestet:
-
-        * Ob man alle Eigenschaften zu Referaten ändern kann
-        * Ob man alle Eigenschaften zu Unterbereichen ändern kann
-        * Ob man alle Eigenschaften zu Ämtern ändern kann
-
+        Setup and Teardown Funktions are specified in
```
Comment (benjamin-hempel) [inline tests/test_009_aemtAendern.py:9]: *functions

**Comment by Yoshinonxkun** [comment]
Das war von Sphinx, der erstellt beim build automatisch den _sources folder der genau dasselbe beinhaltet wie der Ordner aus dem ich die docu baue...

**Pull request merged**